profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/LanApiLiveTest.java

```diff
@@ -0,0 +1,137 @@
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "live", testName = "LanApiLiveTest")
+public class LanApiLiveTest extends BaseProfitBricksLiveTest {
+   
+   DataCenter dataCenter;
+   Lan testLan;
+   Snapshot testSnapshot;
+  
+   @BeforeClass
+   public void setupTest() {
+      dataCenter = createDataCenter();
+   }
+   
+   @AfterClass(alwaysRun = true)
+   public void teardownTest() {
+      if (dataCenter != null)
+         deleteDataCenter(dataCenter.id());
+   }
+     
+   @Test
+   public void testCreateLan() {
+      assertNotNull(dataCenter);
+            
+      testLan = lanApi().create(
+              Lan.Request.creatingBuilder()
+              .dataCenterId(dataCenter.id())
+              .name("jclouds-lan")
+              .build());
+
+      assertNotNull(testLan);
+      assertEquals(testLan.properties().name(), "jclouds-lan");
+      assertLanAvailable(testLan);
+   }
+   
+
+   @Test(dependsOnMethods = "testCreateLan")
+   public void testGetLan() {
+      Lan lan = lanApi().get(dataCenter.id(), testLan.id());
+
+      assertNotNull(lan);
+      assertEquals(lan.id(), testLan.id());
+   }
+
+   @Test(dependsOnMethods = "testCreateLan")
+   public void testList() {
+      List<Lan> lans = lanApi().list(dataCenter.id());
+
+      assertNotNull(lans);
+      assertFalse(lans.isEmpty());
+      assertEquals(lans.size(), 1);
+   }
+   
+   @Test(dependsOnMethods = "testGetLan")
+   public void testUpdateLan() {
+      assertDataCenterAvailable(dataCenter);
+      
+      api.lanApi().update(
+              Lan.Request.updatingBuilder()
+              .dataCenterId(testLan.dataCenterId())
+              .id(testLan.id())
+              .isPublic(false)
+              .build());
+
+      assertLanAvailable(testLan);
+      
+      Lan lan = lanApi().get(dataCenter.id(), testLan.id());
+      
+      assertEquals(lan.properties().isPublic(), false);
+   }
+   
+   @Test(dependsOnMethods = "testUpdateLan")
+   public void testDeleteLan() {
+      lanApi().delete(testLan.dataCenterId(), testLan.id());
+      assertLanRemoved(testLan);
+   }
+   
+   private void assertLanAvailable(Lan lan) {
+      assertPredicate(new Predicate<String>() {
+         @Override
+         public boolean apply(String args) {
+            String[] params = args.split(",");
+            Lan lan = lanApi().get(params[0], params[1]);
+            
+            if (lan == null || lan.metadata() == null)
+               return false;
+            
+            return lan.metadata().state() == State.AVAILABLE;
+         }
+      }, complexId(lan.dataCenterId(), lan.id()));
+   }
+   
+   private void assertLanRemoved(Lan lan) {
+      assertPredicate(new Predicate<String>() {
+         @Override
+         public boolean apply(String args) {
+            String[] params = args.split(",");
+            return lanApi().get(params[0], params[1]) == null;
+         }
+      }, complexId(lan.dataCenterId(), lan.id()));
+   }
```

Comment:
Same thing here about this ugly complex id thing. Get rid of this pattern, please.